src/reflect/iter_test.go

```diff
@@ -0,0 +1,416 @@
+			i := N(0)
+			for v := range s {
+				if v.Int() != int64(i) {
+					t.Fatalf("got %d, want %d", v.Int(), i)
+				}
+				i++
+				if v.Type() != reflect.TypeOf(i) {
+					j := ValueOf(i)
+					t.Logf("ValueOf(j): %s", j.Type())
+					t.Fatalf("got %s, want %s", v.Type(), reflect.TypeOf(i))
+				}
+			}
+			if i != 4 {
+				t.Fatalf("should loop four times")
+			}
+		}},
+	}
+	for _, tc := range tests {
+		seq := tc.val.Seq()
+		tc.check(t, seq)
+	}
+}
+
+func TestValueSeq2(t *testing.T) {
+	m := map[string]int{
+		"1": 1,
+		"2": 2,
+		"3": 3,
+		"4": 4,
+	}
+	tests := []struct {
+		name  string
+		val   Value
+		check func(*testing.T, iter.Seq2[Value, Value])
+	}{
+		{"*[4]int", ValueOf(&[4]int{1, 2, 3, 4}), func(t *testing.T, s iter.Seq2[Value, Value]) {
+			i := int64(0)
+			for v1, v2 := range s {
+				if v1.Int() != i {
+					t.Fatalf("got %d, want %d", v1.Int(), i)
+				}
+				i++
+				if v2.Int() != i {
+					t.Fatalf("got %d, want %d", v2.Int(), i)
+				}
+			}
+			if i != 4 {
+				t.Fatalf("should loop four times")
+			}
+		}},
+		{"[4]int", ValueOf([4]int{1, 2, 3, 4}), func(t *testing.T, s iter.Seq2[Value, Value]) {
+			i := int64(0)
+			for v1, v2 := range s {
+				if v1.Int() != i {
+					t.Fatalf("got %d, want %d", v1.Int(), i)
+				}
+				i++
+				if v2.Int() != i {
+					t.Fatalf("got %d, want %d", v2.Int(), i)
+				}
+			}
+			if i != 4 {
+				t.Fatalf("should loop four times")
+			}
+		}},
+		{"[]int", ValueOf([]int{1, 2, 3, 4}), func(t *testing.T, s iter.Seq2[Value, Value]) {
+			i := int64(0)
+			for v1, v2 := range s {
+				if v1.Int() != i {
+					t.Fatalf("got %d, want %d", v1.Int(), i)
+				}
+				i++
+				if v2.Int() != i {
+					t.Fatalf("got %d, want %d", v2.Int(), i)
+				}
+			}
+			if i != 4 {
+				t.Fatalf("should loop four times")
+			}
+		}},
+		{"string", ValueOf("12语言"), func(t *testing.T, s iter.Seq2[Value, Value]) {
+			next, stop := iter.Pull2(s)
+			defer stop()
+			i := int64(0)
+			for j, s := range "12语言" {
+				v1, v2, ok := next()
+				if !ok {
+					t.Fatalf("should loop four times")
+				}
+				if v1.Int() != int64(j) {
+					t.Fatalf("got %d, want %d", v1.Int(), j)
+				}
+				if v2.Interface() != s {
+					t.Fatalf("got %v, want %v", v2.Interface(), s)
+				}
+				i++
+			}
+			if i != 4 {
+				t.Fatalf("should loop four times")
+			}
+		}},
+		{"map[string]int", ValueOf(m), func(t *testing.T, s iter.Seq2[Value, Value]) {
+			copy := maps.Clone(m)
+			for v1, v2 := range s {
+				v, ok := copy[v1.String()]
+				if !ok {
+					t.Fatalf("unexpected %v", v1.String())
+				}
+				if v != v2.Interface() {
+					t.Fatalf("got %v, want %d", v2.Interface(), v)
+				}
+				delete(copy, v1.String())
+			}
+			if len(copy) != 0 {
+				t.Fatalf("should loop four times")
+			}
+		}},
+		// {"func", ValueOf(func(f func(int, int) bool) {
+		// 	for i := range 4 {
+		// 		f(i, i+1)
+		// 	}
+		// }), func(t *testing.T, s iter.Seq2[Value, Value]) {
```

Comment:
Are these supposed to be commented?

Reply:
TinyGo reflect doesn't support functions so I disabled those tests.